frog/imports/api/sessions.js

```diff
@@ -140,8 +140,14 @@ export const updateOpenActivities = (
 ) => {
   if (Meteor.isServer) {
     Sessions.update(sessionId, { $set: { state: 'WAITINGFORNEXT' } });
+    Sessions.findOne(sessionId).openActivities.forEach(activityId =>
+      Activities.update(activityId, { $set: { actualClosingTime: new Date() } })
+    );
     openActivities.forEach(activityId => {
       Meteor.call('dataflow.run', 'activity', activityId, sessionId);
+      Activities.update(activityId, {
+        $set: { actualStartingTime: new Date() }
```

Comment:
I think that this would be a problem. If an activity runs in parallel to several other activities, then the `actualStartingTime` will be overwritten and only the last starting time will remain. There must be a way to do an update query that does not overwrite.

Reply:
Good point, my latest push fixes this - should be quite robust. Of course, logic has to be revised if we have flexible time in the future, but for now seems to work.